File: Benchmarks.Tests/OptorTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Pidgin;
using Sprache;
using static Benchmarks.Patterns.OptorBenchmarks;

namespace Benchmarks {
	[TestClass]
	public class OptorTests {
		[TestMethod]
		public void Pidgin() {
			Result<Char, String> result;

			result = pidgin.Parse("Hello");
			Assert.IsTrue(result.Success);
			Assert.AreEqual("Hello", result.Value);

			result = pidgin.Parse("World");
			Assert.IsFalse(result.Success);
		}

		[TestMethod]
		public void Sprache() {
			IResult<IOption<String>> result;

			result = sprache.TryParse("Hello");
			Assert.IsTrue(result.WasSuccessful);
			Assert.IsTrue(result.Value.IsDefined);
			Assert.AreEqual("Hello", result.Value.Get());

			result = sprache.TryParse("World");
			Assert.IsTrue(result.WasSuccessful);
			Assert.IsFalse(result.Value.IsDefined);
		}
	}
}

File: Patterns.Tests/OptorTests.fs
namespace Patterns

open System
open Stringier.Patterns
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type OptorTests() =

    [<TestMethod>]
    member _.``consume`` () =
        let pattern = maybe "Hello"

        ResultAssert.Captures("Hello", pattern.Consume("Hello world!"))
        ResultAssert.Captures("", pattern.Consume("Goodbye world!"))

    [<TestMethod>]
    member _.``neglect`` () =
        let pattern = not (maybe "Hello")
        ResultAssert.Captures("", pattern.Consume("Hello"))
        ResultAssert.Captures("World", pattern.Consume("World"))